### PR TITLE
Rename login_whitelist_ip config to login_allowlist_ip

### PR DESCRIPTION
**Pull request by tsteur**
Not needed in developer changelog as we automatically migrate existing setting and if that doesn't work then we still support previous settings for BC (important as it is a security related feature, not always do config changes work eg when they forget to distribute it across multiple servers etc). Also it's not a developer setting but just a regular user setting.

Also renames `login_whitelist_apply_to_reporting_api_requests` to `login_allowlist_apply_to_reporting_api_requests`

once merged will need to update the FAQ (I'll do this)

### CODE REVIEW

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: config/global.php [hunk 1]
```diff
@@ -146,13 +146,16 @@
 
     'Piwik\EventDispatcher' => DI\autowire()->constructorParameter('observers', DI\get('observers.global')),
 
-    'login.whitelist.ips' => function (ContainerInterface $c) {
+    'login.allowlist.ips' => function (ContainerInterface $c) {
```
[the remark this answers is not in the document]
Comment (tsteur) [reply config/global.php:149]: DI configs aren't API currently and AFAIK not documented anywhere etc. They are basically just for us so far

File: plugins/CoreHome/tests/Integration/LoginAllowlistTest.php [hunk 2]
```diff
@@ -60,72 +60,85 @@ public function tearDown(): void
 
     public function test_shouldWhitelistApplyToAPI_shouldBeEnabledByDefault()
     {
-        $this->assertTrue($this->whitelist->shouldWhitelistApplyToAPI());
+        $this->assertTrue($this->allowlist->shouldAllowlistApplyToAPI());
     }
 
     public function test_shouldWhitelistApplyToAPI_canBeDisabled()
     {
         $this->setGeneralConfig('login_whitelist_apply_to_reporting_api_requests', '0');
```
[the remark this answers is not in the document]
Comment (tsteur) [reply plugins/CoreHome/tests/Integration/LoginAllowlistTest.php:68]: I'll commit this separately. I partially left some with whitelist to make sure we still support the previous setting re BC  but will rather add an extra test for it

File: core/API/Request.php [hunk 3]
```diff
@@ -238,10 +238,10 @@ public function process()
             // IP check is needed here as we cannot listen to API.Request.authenticate as it would then not return proper API format response.
             // We can also not do it by listening to API.Request.dispatch as by then the user is already authenticated and we want to make sure
             // to not expose any information in case the IP is not whitelisted.
```
[the remark this answers is not in the document]
Comment (tsteur) [reply core/API/Request.php:240]: will commit it separately

File: config/global.ini.php [hunk 4]
```diff
@@ -432,16 +432,16 @@
...
+; login_allowlist_ip[] = 204.93.240.*
+; login_allowlist_ip[] = 204.93.177.0/24
+; login_allowlist_ip[] = 199.27.128.0/21
+; login_allowlist_ip[] = 2001:db8::/48
+; login_allowlist_ip[] = matomo.org
 
-; By default, if a whitelisted IP address is specified via "login_whitelist_ip[]", the reporting user interface as
+; By default, if a whitelisted IP address is specified via "login_allowlist_ip[]", the reporting user interface as
 ; well as HTTP Reporting API requests will only work for these whitelisted IPs.
 ; Set this setting to "0" to allow HTTP Reporting API requests from any IP address.
```
[the remark this answers is not in the document]
Comment (tsteur) [reply config/global.ini.php:443]: will do 👍

**Comment by tsteur** [comment]
@sgiehl applied the review comments. replaced the word also in few other places but not in places like the validator or so when it requires changes across plugins as it's not as easy to do and needs a separate PR anyway

File: plugins/CoreHome/LoginAllowlist.php [hunk 5]
```diff
@@ -20,15 +20,15 @@
...
  */
-class LoginWhitelist
+class LoginAllowlist
 {
-    public function shouldWhitelistApplyToAPI()
+    public function shouldAllowlistApplyToAPI()
     {
         $general = $this->getGeneralConfig();
-        return !empty($general['login_whitelist_apply_to_reporting_api_requests']);
+        return !empty($general['login_allowlist_apply_to_reporting_api_requests']) || !empty($general['login_whitelist_apply_to_reporting_api_requests']);
```
Comment (sgiehl) [inline plugins/CoreHome/LoginAllowlist.php:28]: The fallback to the old name might actually not work as expected in all cases as the default value of those setting is `1`.
Lets say someone does not configure `login_allowlist_apply_to_reporting_api_requests`, but uses `login_whitelist_apply_to_reporting_api_requests = 0`. That would actually not work, as `login_allowlist_apply_to_reporting_api_requests` defaults to `1`.

I guess that might be fine, as `login_allowlist_apply_to_reporting_api_requests` will be set on update if needed, but actually I think we could remove the fallback here then...

**Comment by sgiehl** [comment]
Seems you added the screenshot directly, and not to LFS...

**Comment by tsteur** [comment]
fixed

**Pull request merged**